commit 15a3018bda8ee92393671ceaf272be418e95a81f
Author: Lennart Poettering <lennart@poettering.net>
Date:   Tue Jan 12 23:01:24 2010 +0100

    udev: don't forget to unref devices we are not interested in
    
    https://bugzilla.redhat.com/show_bug.cgi?id=552932

diff --git a/src/modules/module-udev-detect.c b/src/modules/module-udev-detect.c
index 1b1e9c1..7508d9a 100644
--- a/src/modules/module-udev-detect.c
+++ b/src/modules/module-udev-detect.c
@@ -416,8 +416,10 @@ static void monitor_cb(
         goto fail;
     }
 
-    if (!path_get_card_id(udev_device_get_devpath(dev)))
+    if (!path_get_card_id(udev_device_get_devpath(dev))) {
+        udev_device_unref(dev);
         return;
+    }
 
     process_device(u, dev);
     udev_device_unref(dev);
